frontend/src/components/StatsCard.js

```diff
@@ -0,0 +1,207 @@
+    return retval.toFixed(2);
+  } else {
+    return "-";
+  }
+};
+
+const getDiff = (a, b) => {
+  if (isNaN(a) || isNaN(b)) {
+    return "-";
+  } else {
+    return Number(a) - Number(b);
+  }
+};
+
+const getSign = (a) => {
+  if (isNaN(a)) return "-";
+  return Number(a) >= 0 ? true : false;
+};
+const StatsCard_ = ({ className, label, netLabel, labelKey, timeRange }) => {
+  const { nftCache } = useNFTs();
+
+  const [nftData, setData] = useState();
+
+  useEffect(() => {
+    if (nftCache.data[TIME_PERIOD.current][labelKey][timeRange]) {
+      const cacheData = nftCache.data[TIME_PERIOD.current][labelKey][timeRange];
+      const prevCacheData =
+        nftCache.data[TIME_PERIOD.previous][labelKey][timeRange];
+      const valueChange = getChange(cacheData.amount, prevCacheData.amount);
+      const share = isNaN(cacheData.percentage)
+        ? "-"
+        : Number(cacheData.percentage).toFixed(2);
+      const shareChange = getDiff(
+        cacheData.percentage,
+        prevCacheData.percentage
+      );
+
+      setData({
+        dimension: labelKey === "values" ? "Eth" : "#",
+        isValueIncrease: getSign(valueChange),
+        isShareIncrease: getSign(shareChange),
+        valueChange,
+        shareChange,
+        share,
+        value:
+          labelKey === "values"
+            ? getEthValue(cacheData.amount)
+            : nFormatter(cacheData.amount, 3),
+      });
+    }
+  }, [nftCache?.data, nftCache.isLoading, labelKey, timeRange]);
+  if (nftCache.isLoading || !nftData) return "";
+
+  return (
+    <Stack className={className}>
+      <Box
+        w="full"
+        borderTopRadius="inherit"
+        fontWeight="600"
+        bgColor="gray.200"
+        px={4}
+        textAlign="center"
+      >
+        {label}
+      </Box>
+      <SimpleGrid
+        columns={2}
+        justifyItems="center"
+        textAlign="center"
+        h="100%"
+        alignContent="center"
+      >
+        <Box
+          w="100%"
+          fontSize="1.125rem"
+          borderStyle="dashed"
+          borderRightWidth="3px"
+          borderRightColor="gray.300"
+          //   alignItems="center"
+          h="100%"
+        >
+          <Link
+            textDecorationLine="underline"
+            textDecorationStyle="dashed"
+            textUnderlineOffset={2}
+            boxDecorationBreak="slice"
+            textDecorationThickness="1px"
+          >
+            {nftData.value}
+          </Link>
+          <Text pl={2} display="inline-block">
+            {nftData.dimension}
+          </Text>
+        </Box>
+        <Stack
+          w="100%"
+          direction="row"
+          fontSize="1.125rem"
+          placeContent="center"
+          alignItems="center"
+        >
+          {nftData.isValueIncrease && <TriangleUpIcon color="suggested.900" />}
+          {!nftData.isValueIncrease && <TriangleDownIcon color="unsafe.900" />}
+          <Text
```

Comment:
Add `id` attributes to each of these tags so that we can understand easily in the code and in the DOM what the nodes represent.

Maybe some comments could help as well, to map the components to the picture of the card.